examples/grpc/greeter/client.cpp

```diff
@@ -0,0 +1,84 @@
+#include "greeter.grpc.fb.h"
+#include "greeter_generated.h"
+
+#include <grpc++/grpc++.h>
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+class GreeterClient {
+ public:
+  GreeterClient(std::shared_ptr<grpc::Channel> channel)
+    : stub_(Greeter::NewStub(channel)) {}
+
+  std::string SayHello(const std::string& name) {
+    auto name_offset = mb_.CreateString(name);
+    auto request_offset = CreateHelloRequest(mb_, name_offset);
+    mb_.Finish(request_offset);
+    auto request_msg = mb_.ReleaseMessage<HelloRequest>();
+
+    flatbuffers::grpc::Message<HelloReply> response_msg;
+
+    grpc::ClientContext context;
+
+    auto status = stub_->SayHello(&context, request_msg, &response_msg);
+    if (status.ok()) {
+      const HelloReply* response = response_msg.GetRoot();
+      return response->message()->str();
+    } else {
+      std::cerr << status.error_code() << ": " << status.error_message()
+                << std::endl;
+      return "RPC failed";
+    }
+  }
+
+  template <class F>
+  void SayManyHellos(const std::string& name, int num_greetings, F&& callback) {
```

Comment:
if its not performance sensitive, maybe use `std::function` for callbacks? or maybe no callback at all, since in an example, simple straightline code to follow is more important than modularity/factoring concerns.